internal/pkg/docker/dockerfile/dockerfile.go

```diff
@@ -0,0 +1,152 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package dockerfile
+
+import (
+	"bufio"
+	"errors"
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/spf13/afero"
+)
+
+const exposeRegexPattern = `(\d+)(\/(\w+))?` // port and optional protocol, at least 1 time on a line
+
+var (
+	errCouldntParseDockerfilePort = errors.New("parse port from EXPOSE")
+)
+
+type Dockerfile interface {
+	GetExposedPorts() []uint16
+}
+
+type PortConfig struct {
+	Port      uint16
+	Protocol  string
+	RawString string
+}
+
+type Config struct {
+	ExposedPorts []PortConfig
+	parsed       bool
+	path         string
+
+	fs afero.Fs
+}
+
+func NewConfig(fs afero.Fs, path string) *Config {
+	return &Config{
+		fs:     fs,
+		path:   path,
+		parsed: false,
+	}
+}
+
+func (df *Config) GetExposedPorts() []uint16 {
+	if !df.parsed {
+		df.parse()
```

Comment:
What happens if this errors? Should the method signature be `GetExposedPorts() ([]uint16, error)` instead?

Reply:
I like this - I think it'll help clean up that awkward if ports[0] == 0 case in app_init.

Reply:
Good question--since this is mostly a form of sugar, errors in dockerfile parsing should only be surfaced if they're severe, since if this package fails entirely we will prompt the user to enter a port (default 80). I'll ask for opinions in chat but perhaps @kohidave or @iamhopaul123 can weigh in?